frontend_tests/node_tests/recent_senders.js

```diff
@@ -310,3 +313,38 @@ test("process_stream_message", () => {
         true,
     );
 });
+
+test("process_pms", () => {
+    const sender1 = 1; // Current user id
+    const sender2 = 2;
+    const sender3 = 3;
+
+    const user_ids_string = "2,3,4";
+    rs.process_private_message({
+        to_user_ids: user_ids_string,
+        sender_id: sender2,
+        id: 1,
+    });
+    rs.process_private_message({
+        to_user_ids: user_ids_string,
+        sender_id: sender3,
+        id: 2,
+    });
+    rs.process_private_message({
+        to_user_ids: user_ids_string,
+        sender_id: sender1,
+        id: 3,
+    });
+
+    // Recent topics displays avatars in the opposite order to this since
+    // that was simpler to implement in HTML.
+    assert.deepEqual(rs.get_pm_recent_senders(user_ids_string), {
+        participants: [2, 3, 1],
```

Comment:
Reading this still feels confusing; could we just add a reverse somewhere in the HTML generation code path, and have this function return the users sorted by "most recent participant first"? I feel like there's a significant risk that someone reusing this function, say to have mention typeahead sort by the most recent participant when you're in a group PM conversation, will get it backwards as written :)

Reply:
Done! Did the same for topics as well.